src/lib/utils/size_estimation_utils.hpp

```diff
@@ -12,15 +12,18 @@ namespace opossum {
 template <typename T>
 size_t string_heap_size(const T& string) {
   // Get the default pre-allocated capacity of SSO strings.
-  const auto sso_string_capacity = T("").capacity();
+  const auto sso_string_capacity = T{}.capacity();
 
   if (string.capacity() > sso_string_capacity) {
     // For heap-allocated strings, \0 is appended to denote the end of the string. capacity() is used over length()
     // since some libraries (e.g. llvm's libc++) also over-allocate the heap strings
     // (cf. https://shaharmike.com/cpp/std-string/).
     return string.capacity() + 1;
   }
-  return 0;
+
+  // valgrind is drunk. This could be `return 0`, but valgrind doesn't like it.
+  DebugAssert(string.capacity() - sso_string_capacity == 0, "SSO does not meet expectations");
```

Comment:
`string.capacity() == sso_string_capacity` ?

Reply:
I wanted to convey that this is the exact thing as `0`. No hard feelings about it though.

Reply:
In the sense of "I want to return 0, but can't. Here, it is the same thing".

Reply:
Ack.